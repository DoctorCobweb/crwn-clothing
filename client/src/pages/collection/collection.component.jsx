import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { firestore } from '../../firebase/firebase.utils';

import CollectionItem from '../../components/collection-item/collection-item.component';

import { selectCollection } from '../../redux/shop/shop.selectors';

import {
  CollectionPageContainer,
  CollectionTitle,
  CollectionItemsContainer 
} from './collection.styles.jsx';

export const CollectionPage = ({ collection }) => {

  // HERES HOW YOU HANDLE CLEANUP USING useEffect hook
  // => just a demo of functionality here. not needed for the app.
  useEffect(() => {
    console.log('i am subscribgin')
    const unsubscribeFromCollections = firestore.collection('collections').onSnapshot(snapshot => console.log(snapshot))
    return () => {
      console.log('im unsubbing');
      //cleanup function aka like componentWillUnmount
      unsubscribeFromCollections();
    }
  }, []);

  const { title, items } = collection;

  return (
    <CollectionPageContainer>
      <CollectionTitle>{title.toUpperCase()}</CollectionTitle>
      <CollectionItemsContainer >
      {
        items.map(item => <CollectionItem key={item.id} item={item} />)
      }
      </CollectionItemsContainer>
    </CollectionPageContainer>
  )
};

const mapStateToProps = (state, ownProps) => ({
  collection: selectCollection(ownProps.match.params.collectionId)(state),
});

export default connect(mapStateToProps)(CollectionPage);