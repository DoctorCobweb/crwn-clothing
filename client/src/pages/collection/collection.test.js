import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPage } from './collection.component';

describe('CollectionPage component', () => {
  let wrapper;
  let mockTitle;
  let mockItems;

  beforeEach(() => {
    mockTitle = 'BlahTitle';
    mockItems = [{id:0},{id:1},{id:2},{id:3},{id:4}];
    const mockProps = {
      collection: {
        title: mockTitle,
        items: mockItems,
      }
    }
    wrapper = shallow(<CollectionPage {...mockProps} />);
  });

  it('should render CollectionPage', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render all the collection items', () => {
    expect(wrapper.find('CollectionItemsContainer').children().length).toBe(mockItems.length);
  });

  it('should render collection title properly', () => {
    expect(wrapper.find('CollectionTitle').text()).toBe(mockTitle.toUpperCase());
  });

});