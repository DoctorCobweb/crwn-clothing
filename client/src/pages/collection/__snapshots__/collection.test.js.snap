// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage component should render CollectionPage 1`] = `
"<CollectionPageContainer>
  <CollectionTitle>
    BLAHTITLE
  </CollectionTitle>
  <CollectionItemsContainer>
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
    <Connect(CollectionItem) item={{...}} />
  </CollectionItemsContainer>
</CollectionPageContainer>"
`;
